serde-avro/src/header.rs

```diff
@@ -1,9 +1,28 @@
 use serde;
-use std::collections;
+use std::{collections};
 
-#[derive(Clone, Debug, Deserialize, Serialize)]
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
 pub struct Header {
     pub magic: serde::bytes::ByteBuf,
     pub meta: collections::HashMap<String, serde::bytes::ByteBuf>,
     pub sync: serde::bytes::ByteBuf,
 }
+
+impl Header {
+
+    /// Returns a copy of the given header's name field as a String.
+    pub fn name(&self) -> String {
+        let raw = self.get("name").expect("No name present in Avro schema!");
+        String::from_utf8(raw.to_vec()).expect("Failed to decode name!")
+    }
+
+    /// Returns a copy of the raw bytes stored at the given key
+    /// for the given header.
+    pub fn get(&self, key: &str) -> Option<serde::bytes::ByteBuf> {
```

Comment:
Why do we want to copy? Returning reference seems to be more ergonomic. Also the serde buffer type is internal, would prefer &[u8]